test/unit/specs/store/pool.spec.js

```diff
@@ -1,64 +1,83 @@
-import setup from "../../helpers/vuex-setup"
-import lcdClientMock from "renderer/connectors/lcdClientMock.js"
 import poolModule from "renderer/vuex/modules/pool.js"
 
-const instance = setup()
-
-const mockPool = lcdClientMock.state.pool
-const mockRootState = {
-  connection: {
-    connected: true
-  }
-}
-
 describe(`Module: Pool`, () => {
-  let store
+  let module, state, actions, mutations
+  const pool = {
+    loose_tokens: `100.0000000000`,
+    bonded_tokens: `50.0000000000`
+  }
 
   beforeEach(() => {
-    const test = instance.shallow(null)
-    store = test.store
+    module = poolModule({
+      node: {
+        getPool: jest.fn().mockResolvedValue(pool)
+      }
+    })
+    state = module.state
+    actions = module.actions
+    mutations = module.mutations
+  })
+
+  it(`should have correct defaults`, () => {
+    expect(state.pool).toEqual({})
+    expect(state.loading).toEqual(false)
+    expect(state.loaded).toEqual(false)
+    expect(state.error).toEqual(null)
+  })
 
-    store.commit(`setConnected`, true)
+  it(`should set pool`, () => {
+    mutations.setPool(state, pool)
+    expect(state.pool).toBe(pool)
   })
 
-  it(`should have no pool by default`, () => {
-    expect(store.state.pool.pool).toEqual({})
+  it(`should not query the pool if not connected`, async () => {
+    const commit = jest.fn()
+    const rootState = {
+      connection: {
+        connected: false
+      }
+    }
+
+    await actions.getPool({ state, commit, rootState })
+    expect(commit).not.toHaveBeenCalled()
   })
 
   it(`should query pool`, async () => {
-    await store.dispatch(`getPool`)
-    expect(store.state.pool.pool).toEqual(mockPool)
-  })
+    const commit = jest.fn()
+    const rootState = {
```

Comment:
why not define `rootState` above with `true` as default value and only change to `false` when needed? to avoid dup code.

Reply:
good idea. This way the `rootState` could be changed in one place instead of in many.